cirq-core/cirq/transformers/align_test.py

```diff
@@ -71,6 +71,40 @@ def test_align_left_no_compile_context():
     )
 
 
+def test_align_left_subset_of_operations():
+    q1 = cirq.NamedQubit('q1')
+    q2 = cirq.NamedQubit('q2')
+    tag = "op_to_align"
+    c_orig = cirq.Circuit(
+        [
+            cirq.Moment([cirq.Y(q1)]),
+            cirq.Moment([cirq.X(q2)]),
+            cirq.Moment([cirq.X(q1).with_tags(tag)]),
+            cirq.Moment([cirq.Y(q2)]),
+            cirq.measure(*[q1, q2], key='a'),
+        ]
+    )
+    c_exp = cirq.Circuit(
+        [
+            cirq.Moment([cirq.Y(q1)]),
+            cirq.Moment([cirq.X(q1).with_tags(tag), cirq.X(q2)]),
+            cirq.Moment(),
+            cirq.Moment([cirq.Y(q2)]),
+            cirq.measure(*[q1, q2], key='a'),
+        ]
+    )
+    cirq.testing.assert_same_circuits(
+        cirq.xor_ops_with_tags(
+            cirq.align_left(
+                cirq.xor_ops_with_tags(c_orig, [tag]),
```

Comment:
What is going on here? Is the xor a necessary step for alignment, or just a test element to verify proper behavior of `align_left`?

Reply:
So in the example here, directly doing an `align_left` on the entire circuit would cause both `X(q1)` and `Y(q2)` in moments 2 & 3 (0-based indexing) to move left. But by adding a tag to `X(q1)` and using `xor_ops_with_tags` to toggle tags --> do optimization --> toggle tags again, we are able to do the transformation only the subset of operations that are tagged -- therefore only `X(q1)` aligns to the left but `Y(q2)` stays in-place.